### PR TITLE
 Optional urlPrefix and apiVersion for Platform.createUrl.

### PR DESCRIPTION
**Pull request by MostFrumiousBandersnatch**
This is how we're about to utilize it:
```
 sdk().platform().createUrl(
       this.getRawURI(args),
       {
             addToken: true,
             addServer: true
             urlPrefix: '/rcvideo',
             apiVersion: 'v1'
       }
)
```

### CODE REVIEW

**Comment by kirill-konshin** [comment]
The better idea would be to just add something like `knownPrefixes` property because otherwise you will have to manually run `createUrl` for any request that you send. Not convenient...

**Comment by MostFrumiousBandersnatch** [comment]
Not sure, that it would a better idea.

Thus, we would initialize the SDK object with some `knownPrefixes` and, then call `createUrl` pointing one of them? How? Implicitly, by number? Either explicitly, and face with possible rejections when given values aren't ones among predefined `knownPrefixes`?
That's a responsibility of higher-level code, to provide a necessary abstraction around  `createUrl`.  The SDK itself hardly should be concerned about that.
Nothing would prevent us from wrapping  `createUrl` into few different functions, specifying different options in each one.

**Comment by MostFrumiousBandersnatch** [comment]
I decided to neglect the merging predefined knownPrefixes with passed knownPrefixes, in order to keep code base small (avoiding extra dependencies). Let me know if it's (necessity of merge) really a case.

File: src/platform/Platform.js [hunk 1]
```diff
@@ -135,11 +140,16 @@ Platform.prototype.createUrl = function(path, options) {
...
+        hasHttp = path.indexOf('http://') != -1 || path.indexOf('https://') != -1,
+        alreadyPrefixed = this._knownPrefixes.some(function(prefix) {
+            return path.indexOf(prefix) === 0;
+        });
 
     if (options.addServer && !hasHttp) builtUrl += this._server;
 
-    if (path.indexOf(Platform._urlPrefix) == -1 && !hasHttp) builtUrl += Platform._urlPrefix + '/' + Platform._apiVersion;
+    if (path.indexOf(Platform._urlPrefix) == -1 && !hasHttp && !alreadyPrefixed) {
+        builtUrl += [Platform._urlPrefix, Platform._apiVersion].join('/');
```
Comment (kirill-konshin) [inline src/platform/Platform.js]: Weird code style to add a slash between two strings )

File: src/platform/Platform.js [hunk 2]
```diff
@@ -78,6 +78,10 @@ function Platform(options) {
     /** @private */
     this._client = options.client;
 
+    /** @private */
+    this._knownPrefixes = options.knownPrefixes || Platform._knownPrefixes;
```
Comment (kirill-konshin) [inline src/platform/Platform.js]: That's OK.

File: src/platform/Platform.js (hunk 1 above)
Comment (MostFrumiousBandersnatch) [reply src/platform/Platform.js]: It's just more efficient than string concatenation. Anyway, i can replace it.

**Comment by kirill-konshin** [comment]
All good. Please bump minor version, make a build and update this PR. Also please figure out what's wrong with failed CI on Node 4, maybe we should kick it out of stack.

**Comment by MostFrumiousBandersnatch** [comment]
I have no idea why it fails just on Node 4, being run in Travis. Error seems to be irrelevant to node version, furthermore it run normally on my machine, using the very same version of node.
Anyway brief investigation of the issue reveals something:

[Subscription-spec.js](https://github.com/ringcentral/ringcentral-js/blob/master/src/subscription/Subscription-spec.js#L24-L25)
```
return subscription.setEventFilters(['foo', 'bar'])
```

[Subscription.js](https://github.com/ringcentral/ringcentral-js/blob/master/src/subscription/Subscription.js#L328-L334)
```
Subscription.prototype._getFullEventFilters = function() {
    return this.eventFilters().map(function(event) {
        return this._platform.createUrl(event);
    }.bind(this));
}
```
which generates suspicious URLs:
```
> sdk.platform().createUrl('foo');
'/restapi/v1.0foo
```

Hope, that may give you a hint to fix that issue.

**Comment by kirill-konshin** [comment]
Remove Node 4 from `travis.yml`

**Comment by MostFrumiousBandersnatch** [comment]
Are you sure to remove it? Version of node is hardly  a reason, assuming results of [recent build]( https://travis-ci.org/ringcentral/ringcentral-js/builds/340949934?utm_source=github_status&utm_medium=notification) .

**Pull request merged**